### ISSUE TITLE
[Feature] Autostop definition in YAML?

### ISSUE THREAD

**Issue by jucor** [skypilot-org/skypilot#3953]
Is it possible to add an `autostop` field in the YAML for creating a cluster, please? I do not see any such way in https://github.com/skypilot-org/skypilot/blob/master/docs/source/reference/yaml-spec.rst 

The current way to implement autostop seems to require running a command after-launch. Guess who forgot to run it a month ago and just got hit by a multi-hundred dollar bill from EC2? 💸 💸 💸 

Adding `autostop` in the cluster definition would save me (and probably others) from shooting themselves in the foot again. Thanks !

**Comment by jucor** [comment]
I note that #1435 got hit with a similar need, and suggested another way: setting a default value for autostop. Both of these issue (and my 500 USD wasted EC2 bill 😢 ) show there's a need for it.  
@Michaelvll, do you know if this is possible, please?

**Comment by Michaelvll** [comment]
Good point! We debated for a while whether to include the autostop setting in task YAML, since it is more about the task itself than the config for a cluster. One proposal is to have `autostop_after_idle_minutes:` under the `resources` section for the task YAML. Wdyt?

**Comment by jucor** [comment]
The proposed `autostop_after_idle_minutes` in the YAML file seems, to me, a perfect solution for this need!

I did think about the task vs cluster: while it's true that this is a cluster task, a common (and simple) usage pattern, even used in the [Quickstart](https://skypilot.readthedocs.io/en/latest/getting-started/quickstart.html), is to define a task in a YAML and use `sky launch` on it which automatically provisions a cluster for it. 
To be honest, I'm not even sure how I would create a cluster without a task description (except by using command line arguments, but that's more cumbersome than defining a YAML for a task). That's how easy it is to have ad-hoc clusters :)

As to the wasted bill, ah well, that's part of the risks of using the cloud: we all have been or will be hit by such a mistake once in our job 😬  🤷 You provide powerful tools, you're not responsible for how I (fail to) use it 😆  But I do appreciate your receptivity to helping make it easier!

**Comment by Michaelvll** [comment]
A better interface might be that:
```yaml
resources:
  autostop:
    down: true
    idle_minutes: 10
```

With this it can support autodown as well.

**Comment by jucor** [comment]
Sounds perfect too! Whichever is easiest for you to implement, really.

**Comment by romilbhardwaj** [comment]
Bump - also requested by user on slack.

**Comment by kuza55** [comment]
This would be super useful since it's very easy to forget to set the idle value on the command-line.

**Comment by SeungjinYang** [comment]
Taking a look at this issue. I'll try to implement it using this syntax:
```
resources:
  autostop:
    down: true
    idle_minutes: 10
```

Reference for myself:
- https://github.com/skypilot-org/skypilot/pull/5230/files
- https://github.com/skypilot-org/skypilot/pull/5182
as an example of something similar being done for job controllers
